pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -118,7 +122,7 @@ func (c *APIServiceController) sync(ctx context.Context, syncCtx factory.SyncCon
 		return err
 	}
 
-	err = c.syncAPIServices(ctx, apiServices, syncCtx.Recorder())
+	err = c.syncAPIServices(ctx, enabledApiServices, disabledApiServices, syncCtx.Recorder())
```

Comment:
How about adding a separate method for reconciling `disalbedApiServices` and calling it before  `preconditionForEnabledAPIServices` (line 101)?

Having two separate methods would allow us to reconcile `enabledApiServices` and `disabledApiServices` independently. I think we always want to reconcile both and aggregate the errors. What do you think ?

Reply:
Updated